### ISSUE TITLE
macOS Installer Does Not Recognize FPv2

### ISSUE THREAD

**Issue by rjtucker** [navelpluisje/reasonus-install#2]
Hello.

Thank you for creating ReaSonus.

The installer does not recognize my FPv2.

Universal Control was not running while I attempted to install ReaSonus. However, when I later started Universal Control, it correctly identified my FPv2.

Please see the attached screen shots that offer details of my gear.

Thanks.

Regards,
Jim

---
![Reasonus installer](https://user-images.githubusercontent.com/7249593/170791004-e9d3b454-1dd9-4de3-99e5-731dc7341109.png)
![Screenshot 2022-05-27 at 2 20 28 PM](https://user-images.githubusercontent.com/7249593/170791009-0c211f9e-e59c-46c3-b7ac-043dc953007e.png)
![Screenshot 2022-05-27 at 2 23 04 PM](https://user-images.githubusercontent.com/7249593/170791179-fc770d09-5670-4eb1-96ef-305b628915c9.png)

**Comment by navelpluisje** [comment]
Hi Jim,

FaderPort V2 is not supported yet. The issue is that I do not have an V2. So for now only the FaderPort 8 and 16 are supported. We are now close to release it as a beta version. I can try to dig into the V2. 
Your help would then be welcoming

Cheers,
Erwin

**Comment by rjtucker** [comment]
Hello, Erwin.

Thanks for your message. 

Yes, I will gladly help with V2 support. 

Meanwhile, while we are waiting for ReaSonus to support V2, can you please tell me where I can find help getting regular CSI working with my V2?

Thanks. 

Regards,
Jim

**Comment by navelpluisje** [comment]
This thread on the REAPER forum: https://forum.cockos.com/showthread.php?t=245280

**Comment by rjtucker** [comment]
> For the FaderPort 8 and 16 I primarily focussed on the Studio One functionality, because that is the most extensive. Will that also be preferable for the V2?

Yes, please. That’s a good place to start. 

Thanks for looking into supporting the FPv2. Hopefully it will be easy for you, because it’s simpler than the FP 8 & 16. 

Let me know what I can do to assist you. 

J

**Comment by rjtucker** [comment]
Hello, Erwin. 

Are you ready for me to help with the FPv2 now?

Regards,
Jim

**Comment by navelpluisje** [comment]
Hi Jim,
Will start this week with V2. I'll keep you posted on this.

**Comment by rjtucker** [comment]
Thanks, Erwin. 

I have a little time off right now, so I’m happy to help you when you’re ready.

Regards,
Jim

**Comment by navelpluisje** [comment]
In the attachment is the first setup for the V2 zone and surface files. They need to be installed manually and work with the current v1.1 of CSI. For the FaderPort v2 no additional changes are needed.

I implemented most of the functionality as stated with the buttons including the shift functions.
There are some differences:
* The Pan button houses the Banking. I'm not quite sure it is needeed because it has 1 fader.
* Link, Lock and Flip are not implemented yet. Do you know a nice use-case for these?

The function keys have the next actions attached:
F1: Toggle mixer visible
F2: Show/hide media explorer
F3: Show FX browser window
F4: Toggle show all floating windows

This will be editable. I'm also working on a script to edit these in reaper itself. That is almost done now, but have to make a small addition I think, because the V2 has 4 funtion keys in stead of 8

For the next steps:
* Can you provide me your `reaper-midihw.ini`, which is in the REAPER resource folder?  That will hold the info for me to add the device to the lists in the installer.

I'm also a week of. Will do some stuff, but not in a quick pace.

[FaderPort V2.zip](https://github.com/navelpluisje/reasonus-install/files/8837737/FaderPort.V2.zip)

**Comment by rjtucker** [comment]
Thanks! I will work on this today and send my results. 

J

**Comment by navelpluisje** [comment]
@rjtucker Please keep in mind to set the controller so studio one mode.

**Comment by rjtucker** [comment]
Hello, Erwin.

Thanks for this.

I have attached my reaper-midihw.ini file for your review. I have manually installed the zone and surface files. I will next fire up Reaper and see how it works.

I will think of use cases for Link, Lock, and Flip, if I can.

Best regards,
Jim



––––––––––––––––––––––––––––
Jim Tucker
***@***.*** | 604-842-3456

**Comment by rjtucker** [comment]
> @rjtucker Please keep in mind to set the controller so studio one mode.

Hello, Erwin.

I have put the FPv2 controller into studio one mode, but it mostly does not work with the files you provided.

The play, stop, FF, and Rewind buttons work, as does the Click button. But everything else, including the single fader, are not working.

Please advise.

Thanks.

Regards,
Jim

**Comment by navelpluisje** [comment]
Ah, looks like I used the wrong 'navigators' here. Will change this. The attachments do not popup in Github when repleing to a mail. Can you just past the content iin the mail?

**Comment by navelpluisje** [comment]
And for the buttons underneath the navigation controls, you can check the documentation: https://navelpluisje.github.io/reasonus-faderport/. marker, section, master, click, channel scroll and zoom should work the same

**Comment by rjtucker** [comment]
Hello, Edwin.

Here are the contents of the file that you requested.

Thanks.

Regards,
Jim


[mididevcache]
ia0=
ia1=
ia2=
ia3=
ia4=
in0=Scarlett 6i6 USB
in1=PreSonus FP2
in2=EZbass
in3=EZdrummer 3
in4=EZkeys
it0=1654560254
it1=1654390342
it2=1653698080
it3=1653774294
it4=1654560236
ix0=Focusrite - Scarlett 6i6 USB
ix1=PreSonus - PreSonus FP2
ix2=EZbass
ix3=EZdrummer 3
ix4=EZkeys
oa0=
oa1=
on0=Scarlett 6i6 USB
on1=PreSonus FP2
ot0=1654560254
ot1=1654390342
ox0=Focusrite - Scarlett 6i6 USB
ox1=PreSonus - PreSonus FP2

**Comment by navelpluisje** [comment]
And another attempt. It's not completely there. Need to make changes to some navigation files.
Before pasting into the csi folder, remove the current ones first. Did some name changes.

Btw, you're on a Mac, right?

[FaderPort V2.zip](https://github.com/navelpluisje/reasonus-install/files/8856106/FaderPort.V2.zip)

**Comment by rjtucker** [comment]
Thanks, Edwin. 

Yes, I’m on a Mac. I will look at this today (Vancouver time). 

Regards,
Jim

**Comment by rjtucker** [comment]
> And another attempt. It's not completely there. Need to make changes to some navigation files. Before pasting into the csi folder, remove the current ones first. Did some name changes.
> 
> Btw, you're on a Mac, right?
> 
> [FaderPort V2.zip](https://github.com/navelpluisje/reasonus-install/files/8856106/FaderPort.V2.zip)

﻿Hello, Edwin.

Thanks for your patience. I sent this (below) via email a number of days ago, but somehow it did not show up here on GitHub. I will no longer trust email responses to GitHub because they (apparently) don’t always work. From here on I will only reply directly, here on the site. 

THE GOOD NEWS
The fader works now.
The following buttons work as expected: Play, Rewind, FF, Solo, Mute, Touch, Write, Reset, Prev, Channel, Click, Record

THE BAD NEWS
The following buttons do not do anything: Arm, Bypass, Link, Repeat (Loop), and the blue rotary dial (none of push, turn clockwise, and turn counterclockwise do anything)
The Pan button lights up when I push it, but none of the controls I have tried actually affect the pan of anything
The Scroll button lights up when I push it, but none of the controls I have tried actually scroll anything in any way
The Master button lights up when I push it, but none of the controls do anything after that
The Next button works, but it jumps two tracks for each push: for example, from track 10, pushing Next jumps directly to track 12, skipping track 11 entirely (Prev, on the other hand, works as expected)
The Section button works but when moving left-to-right, it jumps two sections for each push: for example, from section 1, pushing Next jumps directly to section 3, skipping section 2 entirely (Prev, on the other hand, works as expected) 
The Marker button lights up when I push it, but none of the controls do anything after that

Yes, I am on a Mac. I was careful to remove all the old files before moving your new files into the CSI folder. I checked and verified REAPER control surface settings, as well.

What are the next steps?

**Comment by rjtucker** [comment]
> Do you have the controller in Studio one mode? Page 5 in the docs: https://pae-web.presonusmusic.com/downloads/products/pdf/FaderPort_OwnersManual_V2_EN_09222021.pdf
> 
> They use different settings then HUI or MCU in a lot of cases

Hello, Edwin. Yes, Studio One mode uses different settings than HUI, MCU, and other modes, in many cases.

My device is, indeed, in Studio One mode.

What do you suggest as next steps?

J

**Comment by navelpluisje** [comment]
Hmmmmm, That's annoying.  The issue is that I'm not able to debug this in a proper way. I'll give it a try tomorrow.

**Comment by rjtucker** [comment]
> Hmmmmm, That's annoying. The issue is that I'm not able to debug this in a proper way. I'll give it a try tomorrow.

No doubt. If it helps, we could do a Zoom meeting in which you could remote-control my setup. I believe you're in Europe (and I'm in Vancouver) but I'd be willing to find a time that works for us both, if that would help you.

**Comment by navelpluisje** [comment]
Oversaw something in my agenda, but a zoom session sounds good. What about T§hursday around this time?

**Comment by rjtucker** [comment]
> Oversaw something in my agenda, but a zoom session sounds good. What about Thursday around this time?

Sounds good, Edwin. Let’s meet 16 Jun 2022 at 1 p.m. Pacific Standard Time (North America) which is 20:00 UTC. 

I’ve put it in my calendar and set a reminder for myself.

**Comment by navelpluisje** [comment]
Cool. Will also add it to my agenda

**Comment by rjtucker** [comment]
Do you want to send a Zoom invitation? (I normally use Google Meet when I host, but if you prefer using your Zoom account that’s fine, too.)

**Comment by navelpluisje** [comment]
Google meet is also fine with me. Not even sure if I have a zoom account

**Comment by rjtucker** [comment]
Great. I will send Google Meet details offline, to the email address I found on your web site. 

(Minor, minor detail: earlier I referred to “Pacific Standard Time,” but currently we are following Pacific Daylight Time, so technically I was incorrect. In any event, let’s just meet at 20h UTC on Thursday, and forget my inaccurate conversion to local Vancouver time.)

**Comment by rjtucker** [comment]
Hello, Edwin.

Thanks for your patience. I caught a virus and was out of the loop for a few days.

THE GOOD NEWS
The fader works now.
The following buttons work as expected: Play, Rewind, FF, Solo, Mute, Touch, Write, Reset, Prev, Channel, Click, Record

THE BAD NEWS
The following buttons do not do anything: Arm, Bypass, Link, Repeat (Loop), and the blue rotary dial (none of push, turn clockwise, and turn counterclockwise do anything)
The Pan button lights up when I push it, but none of the controls I have tried actually affect the pan of anything
The Scroll button lights up when I push it, but none of the controls I have tried actually scroll anything in any way
The Master button lights up when I push it, but none of the controls do anything after that
The Next button works, but it jumps two tracks for each push: for example, from track 10, pushing Next jumps directly to track 12, skipping track 11 entirely (Prev, on the other hand, works as expected)
The Section button works but when moving left-to-right, it jumps two sections for each push: for example, from section 1, pushing Next jumps directly to section 3, skipping section 2 entirely (Prev, on the other hand, works as expected) 
The Marker button lights up when I push it, but none of the controls do anything after that

Yes, I am on a Mac. I was careful to remove all the old files before moving your new files into the CSI folder. I checked and verified REAPER control surface settings, as well.

What are the next steps?
